File: src/lecture/Lecture.java
package lecture;

import java.util.ArrayList;
import java.util.List;

import Enums.LectureType;
import Enums.Term;
import IDs.LectureID;

//Serdar Alsan	150120034 
public class Lecture {

	private LectureID id;
	private String name;
	private LectureType lectureType;
	private int credit;
	private List<LectureSession> sessions;
	private Lecture prerequisite;
	private int quota;
	private Term term;


	public Lecture(LectureID id, String name, LectureType lectureType, int credit, List<LectureSession> sessions,
			Lecture prerequisite, int quota, Term term) {
		super();

		this.id = id;
		this.name = name;
		this.lectureType = lectureType;
		this.credit = credit;
		this.sessions = sessions;

		for (LectureSession ls : this.sessions) {
			ls.setLecture(this);
		}

		this.prerequisite = prerequisite;
		this.quota = quota;
		this.term = term;

		if (sessions == null) {
			this.sessions = new ArrayList<LectureSession>();
		}
	}

	// Creating get and set methods for variables
	public String getID() {
		return id.getID();
	}

	public String getName() {
		return name;
	}

	public LectureType getLectureType() {
		return lectureType;
	}

	public int getCredit() {
		return credit;
	}

	public List<LectureSession> getSessions() {
		return sessions;
	}

	public Lecture getPrerequisite() {
		return prerequisite;
	}

	public LectureID getId() {
		return id;
	}

	public void setId(LectureID id) {
		this.id = id;
	}

	public void setSessions(List<LectureSession> sessions) {
		this.sessions = sessions;
	}

	public void setPrerequisite(Lecture prerequisite) {
		this.prerequisite = prerequisite;
	}

	public int getQuota() {
		return quota;
	}

	public void setName(String name) {
		this.name = name;
	}

	public void setLectureType(LectureType lectureType) {
		this.lectureType = lectureType;
	}

	public void setCredit(int credit) {
		this.credit = credit;
	}

	public void setQuota(int quota) {
		this.quota = quota;
	}

	public void removeLectureSession(LectureSession lectureSession) {
		this.sessions.remove(lectureSession);
	}

	public void addPrerequisiteLecture(Lecture lecture) {
		prerequisite = lecture;
	}

	public void removePrerequisitielLecture() {
		this.prerequisite = null;
	}
	
	public Term getTerm() {
		return term;
	}
	
	public void setTerm(Term term) {
		this.term = term;
	}
	


}

File: pythonCode/lecture/Lecture.py
class Lecture(object):
    """ generated source for class Lecture """
    id = LectureID()
    name = str()
    lectureType = LectureType()
    credit = int()
    sessions = List()
    prerequisite = Lecture()
    quota = int()
    term = Term()
    termYear = TermYear()

    def __init__(self, id, name, lectureType, credit, sessions, prerequisite, quota, term, termYear):
        super(Lecture, self).__init__()
        self.id = id
        self.name = name
        self.lectureType = lectureType
        self.credit = credit
        self.sessions = sessions
        for ls in self.sessions:
            ls.setLecture(self)
        self.prerequisite = prerequisite
        self.quota = quota
        self.term = term
        self.termYear = termYear
        if sessions == None:
            self.sessions = ArrayList()

    #  Creating get and set methods for variables
    def getID(self):
        return self.id.getID()

    def getName(self):
        return self.name

    def getLectureType(self):
        return self.lectureType

    def getCredit(self):
        return self.credit

    def getSessions(self):
        return self.sessions

    def getPrerequisite(self):
        return self.prerequisite

    def getId(self):
        return self.id

    def setId(self, id):
        self.id = id

    def setSessions(self, sessions):
        self.sessions = sessions

    def setPrerequisite(self, prerequisite):
        self.prerequisite = prerequisite

    def getQuota(self):
        return self.quota

    def setName(self, name):
        self.name = name

    def setLectureType(self, lectureType):
        self.lectureType = lectureType

    def setCredit(self, credit):
        self.credit = credit

    def setQuota(self, quota):
        self.quota = quota

    def removeLectureSession(self, lectureSession):
        self.sessions.remove(lectureSession)

    def addPrerequisiteLecture(self, lecture):
        self.prerequisite = lecture

    def removePrerequisitielLecture(self):
        self.prerequisite = None

    def getTerm(self):
        return self.term

    def setTerm(self, term):
        self.term = term

    def getTermYear(self):
        return self.termYear

    def setTermYear(self, termYear):
        self.termYear = termYear